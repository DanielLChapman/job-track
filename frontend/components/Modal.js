import React from "react";
import { CSSTransition } from "react-transition-group";

function Modal(props) {
    let t = props.classes;
    return (
        <>
            <div
                className="blur-layer"
                onClick={() => {
                    props.closeFunc(false);
                }}
            ></div>
            <div className={` modal-window ${t} `}>
                <button
                    type="button"
                    className="close-modal"
                    onClick={() => {props.closeFunc(false);}}
                >
                    X
                </button>
                {props.children}
            </div>
        </>
    );
}

export default Modal;
