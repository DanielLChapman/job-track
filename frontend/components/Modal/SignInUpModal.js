import React, { useState } from "react";
import SignIn from "../SignIn";
import SignUp from "../SignUp";

function SignInUpModal({ user, closeFunc }) {

    const [inUp, setInUp] = useState("In");
    return (
        <>
            <div className="header">
                {inUp === "In" && (
                    <>
                        <h1 className="modal-title">Sign In</h1>
                        <button
                            type="button"
                            className="sign-in-up-selector-button"
                            onClick={() => {
                                setInUp("Up");
                            }}
                        >
                            Sign Up
                        </button>
                    </>
                )}
                {inUp === "Up" && (
                    <>
                        <h1 className="modal-title">Sign Up</h1>
                        <button
                            type="button"
                            className="sign-in-up-selector-button"
                            onClick={() => {
                                setInUp("In");
                            }}
                        >
                            Sign In
                        </button>
                    </>
                )}
            </div>
            <section className="modal-content hide-form-title">
                {inUp === "In" && (
                    <SignIn closeFunc={closeFunc} closeValue={false} />
                )}
                {inUp === "Up" && <SignUp />}
            </section>
        </>
    );
}

export default SignInUpModal;
