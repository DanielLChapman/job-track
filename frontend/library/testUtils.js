import casual from 'casual';

casual.seed(231313);

const fakeUser = (overrides) => ({
    __typename: 'User',
    id: '4234',
    name: casual.name,
    email: casual.email,
    ...overrides,
  });

  export {
    fakeUser
  }