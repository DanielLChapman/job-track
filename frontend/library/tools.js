export const convertNum = (num) => {
    let t = num.toString();
    if (t.length < 3) {
        t = "$" + t
        return t;
    }
    let result = [];
    let count = 1;
    for (let i = t.length - 1; i >= 0; i--) {
        result.unshift(t[i]);
        
        if (count % 3 === 0 && i !== 0) {
            result.unshift(',')
        }
        count++;
    }

    result.unshift('$')

    return result.join('');

}