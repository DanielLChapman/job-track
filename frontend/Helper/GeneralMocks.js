
import { CURRENT_USER_QUERY } from '../components/User';
import { fakeUser } from '../library/testUtils';

const fakeUserObj = fakeUser();
const notSignedInMocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: null } },
    },
  ];
  

  const signedInMocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: fakeUser() } },
    },
  ];

  export {
    notSignedInMocks,
    signedInMocks,
}