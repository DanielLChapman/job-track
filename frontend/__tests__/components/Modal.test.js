import {act, screen, render} from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';

import {signedInMocks, notSignedInMocks} from '../../Helper/GeneralMocks'
import wait from 'waait';
import { fakeUser } from '../../library/testUtils';
import Modal from '../../components/Modal';
import userEvent from '@testing-library/user-event';

const user = fakeUser();

describe('It tests the accuracy of modals and switching', () => {
    it ('With no props, it displays the inner text', async () => {
        const { container, debug} = render(
            <MockedProvider >
                <Modal>
                    Find Me
                </Modal>
            </MockedProvider>
        )

        

        expect(container).toMatchSnapshot();

        expect(await screen.findByText('Find Me')).toBeVisible();
    })

    it ('With innertext props, it displays the inner text', async () => {
        const { container, debug} = render(
            <MockedProvider >
                <Modal  modalContent="innertext">
                    Find Me
                </Modal>
            </MockedProvider>
        )

        

        expect(container).toMatchSnapshot();

        expect(await screen.findByText('Find Me')).toBeVisible();
    })

    it ('With createjob props, it displays "Please Log In" without a user', async () => {
        const { container, debug} = render(
            <MockedProvider >
                <Modal  modalContent="createJob">
               
                </Modal>
            </MockedProvider>
        )

        expect(await screen.findByText('Please Log In')).toBeVisible();
    })

    it ('With createjob props, it displays the createJob form if logged in', async () => {
        const { container, debug} = render(
            <MockedProvider >
                <Modal  modalContent="createJob" user={user}>
               
                </Modal>
            </MockedProvider>
        )

        expect(await screen.findByText('Create a new Job')).toBeVisible();
    })

    it ('With classes props, it displays the inner text with className found', async () => {
        const { container, debug} = render(
            <MockedProvider >
                <Modal  modalContent="innertext" classes="find-me">
                    Find Me
                </Modal>
            </MockedProvider>
        )

        

        expect(container).toMatchSnapshot();

        expect(await screen.findByText('Find Me')).toBeVisible();

        const element = container.getElementsByClassName('find-me');

        expect(element.length).toBe(1);
    })

    it ('With signInUp props, it displays an error message if logged in', async () => {
        const { container, debug} = render(
            <MockedProvider >
                <Modal  modalContent="signInUp" user={user}>
               
                </Modal>
            </MockedProvider>
        )

        expect(await screen.findByText("You Shouldn't be here!")).toBeVisible();
    })

    it ('With signInUp props, it displays a form with switching options if not logged in', async () => {
        const { container, debug} = render(
            <MockedProvider >
                <Modal  modalContent="signInUp">
               
                </Modal>
            </MockedProvider>
        )

        expect(await screen.findAllByText("Sign In")).toBeInTheDocument;

        const element = container.getElementsByClassName('sign-in-up-selector-button');

        expect(element.length).toBe(1);
    })

    it ('With signInUp props, it switches forms on clicks', async () => {
        const { container, debug} = render(
            <MockedProvider >
                <Modal  modalContent="signInUp">
               
                </Modal>
            </MockedProvider>
        )

        await userEvent.click(screen.getByText('Sign Up'));
        await act(async () => {
            await wait(0);
        })

        expect(await screen.findAllByText("Sign Up")).toBeInTheDocument;
    })


})