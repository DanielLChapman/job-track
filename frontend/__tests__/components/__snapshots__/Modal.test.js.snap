// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It tests the accuracy of modals and switching With classes props, it displays the inner text with className found 1`] = `
<div>
  <div
    class="blur-layer"
  />
  <div
    class=" modal-window find-me "
  >
    <button
      class="close-modal"
      type="button"
    >
      X
    </button>
    Find Me
  </div>
</div>
`;

exports[`It tests the accuracy of modals and switching With innertext props, it displays the inner text 1`] = `
<div>
  <div
    class="blur-layer"
  />
  <div
    class=" modal-window  "
  >
    <button
      class="close-modal"
      type="button"
    >
      X
    </button>
    Find Me
  </div>
</div>
`;

exports[`It tests the accuracy of modals and switching With no props, it displays the inner text 1`] = `
<div>
  <div
    class="blur-layer"
  />
  <div
    class=" modal-window  "
  >
    <button
      class="close-modal"
      type="button"
    >
      X
    </button>
    Find Me
  </div>
</div>
`;
